crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -76,14 +76,14 @@ pub(crate) trait StateWriteExt: StateWrite {
         let bytes = balance
             .try_to_vec()
             .context("failed to serialize balance")?;
-        self.put_raw(balance_storage_key(address.to_str()), bytes);
+        self.put_raw(balance_storage_key(&address.to_string()), bytes);
         Ok(())
     }
 
     #[instrument(skip(self))]
     fn put_account_nonce(&mut self, address: &Address, nonce: Nonce) -> Result<()> {
         let bytes = nonce.try_to_vec().context("failed to serialize nonce")?;
-        self.put_raw(nonce_storage_key(address.to_str()), bytes);
+        self.put_raw(nonce_storage_key(&address.to_string()), bytes);
```

Comment:
In the same light as my other comments, why not pass `address` to `nonce_storage_key` and let it do the bytes-to-hex conversion?